### PR TITLE
Adjust cloudinit to support dhcp

### PR DESCRIPTION
**Pull request by mcbenjemaa**
Towards #29 

I decided to create multiple PRs, for better and faster reviews.

This PR:

- Adds dhcp options in the network-config for cloud-init

### CODE REVIEW

File: pkg/cloudinit/network.go [hunk 1]
```diff
@@ -30,7 +30,17 @@ const (
     eth{{ $index }}:
       match:
         macaddress: {{ $element.MacAddress }}
+      {{- if $element.DHCP4 }}
```
[the remark this answers is not in the document]
Comment (mcbenjemaa) [reply pkg/cloudinit/network.go]: done, you can check

**Comment by mcbenjemaa** [comment]
Tested ✅

File: pkg/cloudinit/network_test.go
Comment (avorima) [inline pkg/cloudinit/network_test.go:1]: Please remove the `InvalidNetworkConfigMalformedIp` test case. It is duplicated with `InvalidNetworkConfigMalformedIP`

**Review by avorima — CHANGES_REQUESTED** [review]

File: pkg/cloudinit/network_test.go
Comment (mcbenjemaa) [reply pkg/cloudinit/network_test.go:1]: right

**Review by avorima — APPROVED** [review]

**Pull request merged**